### PR TITLE
Updates ghc version to `9.0.2` and stackage version to `lts-19.6`.

### PR DESCRIPTION
**Pull request by rimbi**
## Purpose

Updates ghc version to `9.0.2` and stackage version to `lts-19.6`.

## Changes

- Updates dependencies
- Fixes warnings and errors.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: http2-client-grpc/src/Network/GRPC/Client.hs [hunk 1]
```diff
@@ -307,11 +308,11 @@ sendSingleMessage rpc msg encoding flagMod conn connectionFlowControl stream str
...
-            then
-                sendData conn stream flagMod dat
-            else do
-                sendData conn stream id (ByteString.take got dat)
-                goUpload (ByteString.drop got dat)
+                then
+                    sendData conn stream flagMod dat
+                else do
+                    sendData conn stream id (ByteString.take got dat)
+                    goUpload (ByteString.drop got dat)
```
Comment (abizjak) [inline http2-client-grpc/src/Network/GRPC/Client.hs]: I don't get why this was changed. It's just a style change, no better, no worse, hence it should not be done.

I'm guessing the previous style is consistent in the project.

File: bench-tool/src/BenchTool.hs [hunk 2]
```diff
@@ -18,7 +19,7 @@ import Network.GRPC.HTTP2.Encoding as Encoding
 
 -- client
 import Network.GRPC.Client (open, singleRequest, streamReply, streamRequest, Timeout(..), StreamDone(..), CompressMode(..))
-import qualified Network.HTTP2.Client as Client
+import qualified "http2-client" Network.HTTP2.Client as Client
```
Comment (abizjak) [inline bench-tool/src/BenchTool.hs:22]: What is the reason for using the package imports extension?

Why is it needed now, but was not needed before?

We generally don't use this extension, so an argument must be made as to why it is necessary.

File: http2-client-grpc/src/Network/GRPC/Client.hs (hunk 1 above)
Comment (rimbi) [reply http2-client-grpc/src/Network/GRPC/Client.hs]: The compiler does not agree with you unfortunately 😉

File: bench-tool/src/BenchTool.hs (hunk 2 above)
Comment (rimbi) [reply bench-tool/src/BenchTool.hs:22]: There was a name conflict, I don't know yet why it fails now while it did not with the older compiler.

File: http2-client-grpc/src/Network/GRPC/Client.hs (hunk 1 above)
Comment (rimbi) [reply http2-client-grpc/src/Network/GRPC/Client.hs]: I am rechecking...

Comment (rimbi) [reply http2-client-grpc/src/Network/GRPC/Client.hs]: Interestingly, it does not complain now, reverting...

File: bench-tool/src/BenchTool.hs (hunk 2 above)
Comment (abizjak) [reply bench-tool/src/BenchTool.hs:22]: The cause is that in version 3 of this package https://hackage.haskell.org/package/http2-3.0.3/docs/Network-HTTP2-Client.html the `Client` module was added. And LTS-19.6 has version 3, and it had version <3 in the previous LTS we used.

File: stack-19.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+- http2-grpc-types
+- http2-grpc-proto-lens
+- http2-grpc-proto3-wire
+- warp-grpc
+- http2-client-grpc
+- bench-tool
+extra-deps:
+- proto3-wire-1.2.2
+- git: https://github.com/Concordium/http2-client.git
+  commit: d853c330b25fcb4724b2fe46f98c8d18560010a1
```
Comment (abizjak) [inline stack-19.yaml]: I think this PR is OK, but we need to update this commit hash after the http2-client one is resolved.

Comment (rimbi) [reply stack-19.yaml]: updated the reference

**Review by abizjak — APPROVED** [review]

**Pull request merged**